promise_button.js

```diff
@@ -5,18 +5,26 @@
  * @name ngPromiseButton.directive:promiseButton
  * @description promise-button attribute expect function call. when button is pressed, function is called and button displays progress and result of the promise.
  * # promiseButton
+ * 
+ * Added functionality:
+ *  handle ngResource promise and null/empty promises
+ *  attribute to submit form - if a promise is supplied the button turns into type="submit"
+ *  attributes for button success/error class - defaults to btn-success/btn-danger
+ *  attribute to revert to original class after revert period
+ *  classes for span elements & use fa-spin class
+ *  added dependency injection annotations
  */
 angular.module('ngPromiseButton', [])
-  .directive('promiseButton', function ($compile, $timeout) {
+  .directive('promiseButton', ['$compile', '$timeout' , function ($compile, $timeout) {
```

Comment:
I don't like messing the code with build related overhead like this. This can be done build-time using for example https://github.com/olov/ng-annotate